### PR TITLE
Results to research study

### PR DESCRIPTION
**Pull request by lelevine**
This is a conversion of the trialscope results to a searches bundle of FHIR research studies. This branch does not return the bundle - it creates the FHIR results and prints them to the console - so it will still work with the current front end. There still isn't a good place to store the inclusion/exclusion trial criteria. It is currently stored in the name attribute of the enrollment attribute's Group resource.

### CODE REVIEW

File: src/research-study.ts [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+  enrollment?: Reference[];
+  sponsor?: Reference;
+  principalInvestigator?: Reference;
+  site?: Reference[];
+  contained?: (Group | Location | Organization | Practitioner)[];
+
+  constructor(trial: TrialScopeTrial, id: number) { // the ridiculous number of if statements are to account for empties being returned from trialscope - if there's a better eay, please let me know - also empty takes many forms ("", "[]", null, etc.) and I don't actually know what they are for each trialscope attribute and I'm not sure how to find out other than trial and error
+    this.id = String(id);
+    if (trial.nctId) {
+      this.identifier = [{use: "official", system: "http://clinicaltrilas.gov", value: trial.nctId}];
```
Comment (zlister) [inline src/research-study.ts:127]: I think clinicaltrials.gov was misspelled here

Comment (lelevine) [reply src/research-study.ts]: Thanks for catching that! Just pushed the fix

File: src/trialscope.ts [hunk 2]
```diff
@@ -42,9 +178,19 @@ export class TrialScopeQuery {
...
+  /**
+   * The fields that should be returned within the individual trial object.
+   */
+  trialFields = [
+    'nctId', 'title', 'officialTitle', 'conditions', 'keywords', 'gender',
+    'description', 'detailedDescription', 'criteria', 'sponsor',
+    'overallContactName', 'overallContactPhone', 'overallContactEmail',
+    'overallOfficialName', 'overallStatus', 'armGroups', 'phase', 'minimumAge',
+    'studyType', 'countries', 'maximumAge'
+  ];
```
Comment (19ngaddam) [inline src/trialscope.ts:193]: Should minimum and maximum age be added to the TrialScopeTrial object

Comment (lelevine) [reply src/trialscope.ts:193]: I don't think they are currently being used for anything, so I guess they don't need to be there.

**Pull request merged**